File: backend/src/models/Payment.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Payment = sequelize.define('Payment', {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
    },
    order_id: {
      type: DataTypes.UUID,
      allowNull: false,
      unique: true,
      references: {
        model: 'orders',
        key: 'id',
      },
    },
    amount: {
      type: DataTypes.DECIMAL(12, 2),
      allowNull: false,
      validate: {
        min: { args: [0], msg: 'Amount must be positive' },
      },
    },
    method: {
      type: DataTypes.ENUM('transfer', 'cash'),
      allowNull: false,
    },
    status: {
      type: DataTypes.ENUM('pending', 'confirmed', 'rejected'),
      defaultValue: 'pending',
      allowNull: false,
    },
    proof_url: {
      type: DataTypes.STRING(500),
      allowNull: true,
    },
    confirmed_by: {
      type: DataTypes.UUID,
      allowNull: true,
      references: {
        model: 'users',
        key: 'id',
      },
    },
    paid_at: {
      type: DataTypes.DATE,
      allowNull: true,
    },
  }, {
    tableName: 'payments',
    timestamps: true,
    underscored: true,
    indexes: [
      { unique: true, fields: ['order_id'] },
      { fields: ['status'] },
    ],
  });

  return Payment;
};

File: mobile/lib/models/payment.dart
class Payment {
  final String id;
  final String orderId;
  final double amount;
  final String method;
  final String status;
  final String? proofUrl;
  final String? confirmedBy;
  final DateTime? paidAt;
  final DateTime? createdAt;

  Payment({
    required this.id,
    required this.orderId,
    required this.amount,
    required this.method,
    required this.status,
    this.proofUrl,
    this.confirmedBy,
    this.paidAt,
    this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'] ?? '',
      orderId: json['order_id'] ?? '',
      amount: double.tryParse(json['amount']?.toString() ?? '0') ?? 0.0,
      method: json['method'] ?? 'transfer',
      status: json['status'] ?? 'pending',
      proofUrl: json['proof_url'],
      confirmedBy: json['confirmed_by'],
      paidAt:
          json['paid_at'] != null ? DateTime.parse(json['paid_at']) : null,
      createdAt:
          json['created_at'] != null
              ? DateTime.parse(json['created_at'])
              : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'order_id': orderId,
      'amount': amount,
      'method': method,
    };
  }
}
